dev/source/docs/ros2-pi.rst

```diff
@@ -95,3 +95,184 @@ The simplest solution is to wire an open telemetry port.
 
 .. figure:: ../images/PiZero2WTelem.png
    :target: ../_images/PiZero2WTelem.png
+
+This allow GPIO13 and GPIO15 (/dev/ttySO) to communicate with ArduPilot.
+
+Cross Compile an application with Docker
+========================================
+
+..
+    Inspired by https://github.com/Ryanf55/ardupilot_ros/blob/docker-deploy-to-pi/docker/Dockerfile
+
+In order to run applications on a Raspberry Pi, we must cross compile them for the ARM architecture.
+`Docker <https://www.docker.com/>`_ allows an easy way to cross compile. Because of the Pi's limited memory, 
+it is not recommended to attempt compiling on target hardware.
+
+First, create a dockerfile like so on your laptop called ``Dockerfile``
+
+.. code-block:: docker
+
+    FROM arm64v8/ros:humble
+
+    RUN --mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
+        --mount=target=/var/cache/apt,type=cache,sharing=locked \
+        apt-get update && apt-get -y --no-install-recommends install \
+            git cmake build-essential
+
+    WORKDIR /ws
+    RUN git clone --branch humble https://github.com/micro-ROS/micro-ROS-Agent.git
```

Comment:
probably want to get only one branch, so add --depth 1 --no-single-branch

Reply:
```suggestion
    RUN git clone  --depth 1 --branch humble https://github.com/micro-ROS/micro-ROS-Agent.git
```
Single branch should be fine for this purpose.